src/Microsoft.AspNetCore.Mvc.Abstractions/ModelBinding/ModelStateDictionary.cs

```diff
@@ -681,12 +681,7 @@ public static bool StartsWithPrefix(string prefix, string key)
                 throw new ArgumentNullException(nameof(key));
             }
 
-            if (StringComparer.OrdinalIgnoreCase.Equals(key, prefix))
```

Comment:
Looking at your hot functions in #4066 top one is StartsWithPrefix

Currently it first does a string equals OrdinalIgnoreCase; then if not found does a StartsWith test.

String equals will only match one entry; so doing it for every entry then doing StartsWith is unnecessary.